lib/widgets/logo_header.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+
+import '../theme/colors.dart';
+import '../utils/context_extensions.dart';
+
+class LogoHeader extends SliverPersistentHeaderDelegate {
+  LogoHeader({this.expandedHeight = 400, required this.onTap});
+
+  VoidCallback onTap;
+  double expandedHeight;
+
+  @override
+  Widget build(BuildContext context, double shrinkOffset, bool overlapsContent) {
+    return Stack(
+      clipBehavior: Clip.none,
+      fit: StackFit.expand,
+      children: [
+        Opacity(
+          opacity: (1 - shrinkOffset / expandedHeight),
+          child: Container(
+            margin: EdgeInsets.only(
+                bottom: clampDouble(expandedHeight * 0.3 - shrinkOffset, 0, double.infinity)),
+            alignment: Alignment.centerLeft,
+            child: Image.network( //TODO: replace placeholder.
+              'https://i.cbc.ca/1.4357834.1542318134!/fileImage/httpImage/image.jpg_gen/derivatives/16x9_780/old-man-senior-cell-phone-hand.jpg',
+              height: expandedHeight * 0.7,
+              width: MediaQuery.of(context).size.width,
+              fit: BoxFit.none,
+            ),
+          ),
+        ),
+        Positioned(
+          top: expandedHeight / 3 - shrinkOffset,
+          left: MediaQuery.of(context).size.width / 4,
+          child: Opacity(
+            opacity: (1 - shrinkOffset / expandedHeight),
+            child: Card(
+              elevation: 10,
+              shape: RoundedRectangleBorder(
+                borderRadius: BorderRadius.circular(34),
+              ),
+              child: Container(
+                height: MediaQuery.of(context).size.width / 2,
+                width: MediaQuery.of(context).size.width / 2,
+                decoration: BoxDecoration(
+                    gradient: ColorsConsts.toPWRGradient, borderRadius: BorderRadius.circular(34)),
+                child: const Center(
+                  child: Text(
+                    'CatchQR',
+                    style: TextStyle(
+                      fontSize: 32,
+                      fontWeight: FontWeight.w500,
+                      color: ColorsConsts.whiteSoap,
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+        Align(
```

Comment:
when it comes to splitting to separate widgets here. It's probably not 100% necessary, but If I were you I would probably split something here, just to simply "add labels" to some part of the code. 

But it's not 100% necessary, it's not that long file. But we can easily separate "logically" separate elements, so I would opt for some slight splitting.

But tbh I wouldn't extract them to separate file. You can split and place them below with private name (with leading underscore). At least that's my style - some may disagree here.

so not forcing anything here, just my thoughts